odoo-openmrs/src/main/java/com/ozonehis/eip/odoo/openmrs/client/OdooClient.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright © 2024, Ozone HIS <info@ozone-his.com>
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.ozonehis.eip.odoo.openmrs.client;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.singletonList;
+import static java.util.Collections.singletonMap;
+
+import com.ozonehis.eip.odoo.openmrs.Constants;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.xmlrpc.XmlRpcException;
+import org.apache.xmlrpc.client.XmlRpcClient;
+import org.apache.xmlrpc.client.XmlRpcClientConfigImpl;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Getter
+@Component
+public class OdooClient {
+
+    @Value("${odoo.baseUrl}")
+    private String url;
+
+    @Value("${odoo.database}")
+    private String database;
+
+    @Value("${odoo.username}")
+    private String username;
+
+    @Value("${odoo.password}")
+    private String password;
+
+    private Integer uid;
+
+    private XmlRpcClientConfigImpl xmlRpcClientConfig;
+
+    private XmlRpcClient client;
+
+    private static final String SERVER_OBJECT_URL = "%s/xmlrpc/2/object";
+
+    private static final String SERVER_COMMON_URL = "%s/xmlrpc/2/common";
+
+    public OdooClient() {}
```

Comment:
Why not use Lombok annotation here `@NoArgsConstructor`

Reply:
Done